packages/db/src/translate/granules.ts

```diff
@@ -1,10 +1,117 @@
+  filePgModel = new FilePgModel(),
+}: {
+  granulePgRecord: PostgresGranuleRecord,
+  collectionPgRecord: PostgresCollectionRecord,
+  knexOrTransaction: Knex | Knex.Transaction,
+  collectionPgModel: CollectionPgModel,
+  pdrPgModel: PdrPgModel,
+  providerPgModel: ProviderPgModel,
+  filePgModel: FilePgModel,
+}): Promise<ApiGranule> => {
+  const collection = collectionPgRecord || await collectionPgModel.get(
+    knexOrTransaction, { cumulus_id: granulePgRecord.collection_cumulus_id }
+  );
+
+  if (granulePgRecord.collection_cumulus_id !== collection.cumulus_id) {
+    throw new ValidationError(`Collection ${collection.cumulus_id} does not match the Granule's Collection ${granulePgRecord.collection_cumulus_id}`);
+  }
+
+  const files = await filePgModel.search(
+    knexOrTransaction, { granule_cumulus_id: granulePgRecord.cumulus_id }
+  );
+  const executionArns = await getExecutionArnsByGranuleCumulusId(
+    knexOrTransaction,
+    granulePgRecord.cumulus_id,
+    1
+  );
+
+  let pdr;
+
+  if (granulePgRecord.pdr_cumulus_id) {
+    pdr = await pdrPgModel.get(
+      knexOrTransaction, { cumulus_id: granulePgRecord.pdr_cumulus_id }
+    );
+  }
+
+  let provider;
+
+  if (granulePgRecord.provider_cumulus_id) {
+    provider = await providerPgModel.get(
+      knexOrTransaction, { cumulus_id: granulePgRecord.provider_cumulus_id }
+    );
+  }
+
+  const apiGranule: ApiGranule = removeNilProperties({
+    beginningDateTime: granulePgRecord.beginning_date_time?.toISOString(),
+    cmrLink: granulePgRecord.cmr_link,
+    collectionId: constructCollectionId(collection.name, collection.version),
+    createdAt: granulePgRecord.created_at?.getTime(),
+    duration: granulePgRecord.duration,
+    endingDateTime: granulePgRecord.ending_date_time?.toISOString(),
+    error: granulePgRecord.error,
+    execution: executionArns[0] ? getExecutionUrlFromArn(executionArns[0].arn) : undefined,
+    files: files.map((file) => ({
+      ...translatePostgresFileToApiFile(file),
+    })),
+    granuleId: granulePgRecord.granule_id,
+    lastUpdateDateTime: granulePgRecord.last_update_date_time?.toISOString(),
+    pdrName: pdr ? pdr.name : undefined,
+    processingEndDateTime: granulePgRecord.processing_end_date_time?.toISOString(),
+    processingStartDateTime: granulePgRecord.processing_start_date_time?.toISOString(),
+    productionDateTime: granulePgRecord.production_date_time?.toISOString(),
+    productVolume: granulePgRecord.product_volume
+      ? granulePgRecord.product_volume : undefined,
+    provider: provider ? provider.name : undefined,
+    published: granulePgRecord.published,
+    queryFields: granulePgRecord.query_fields,
+    status: granulePgRecord.status as GranuleStatus,
+    timestamp: granulePgRecord.timestamp?.getTime(),
+    timeToArchive: granulePgRecord.time_to_archive,
+    timeToPreprocess: granulePgRecord.time_to_process,
+    updatedAt: granulePgRecord.updated_at?.getTime(),
+  }) as ApiGranule;
+
+  return apiGranule;
+};
```

Comment:
This method was pulled from Rds-Phase2 to facilitate the conversion of postgres granule records to api granule objects